File: app/src/main/java/com/aol/philipphofer/logic/Position.java
package com.aol.philipphofer.logic;

public class Position {

    public int row, column, block;

    public Position(int block, int row, int column) {
        this.row = row;
        this.column = column;
        this.block = block;
    }

    @Override
    public String toString() {
        return "[" + block + "][" + row + "][" + column + "]";
    }

    @Override
    public boolean equals(Object o) {
        if(!o.getClass().equals(this.getClass()))
            return false;
        Position p = (Position) o;
        return p.block == this.block && p.row == this.row && p.column == this.column;
    }
}

File: app/src/main/java/com/aol/philipphofer/logic/Position.kt
package com.aol.philipphofer.logic

class Position(var block: Int, var row: Int, var column: Int) {
    override fun toString(): String {
        return "[$block][$row][$column]"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Position

        if (block != other.block) return false
        if (row != other.row) return false
        if (column != other.column) return false

        return true
    }
}